pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -18,70 +18,191 @@ package k8sorchestrator
 
 import (
 	"context"
+	"fmt"
 	"strconv"
 	"sync"
 
+	cnstypes "github.com/vmware/govmomi/cns/types"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+
 	"sigs.k8s.io/vsphere-csi-driver/pkg/common/config"
 	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/logger"
 	k8s "sigs.k8s.io/vsphere-csi-driver/pkg/kubernetes"
 )
 
 var (
-	k8sOrchestratorInstance         *K8sOrchestrator
-	onceFork8sOrchestratorInstance  sync.Once
-	featureStatesConfigMapName      string
-	featureStatesConfigMapNamespace string
+	k8sOrchestratorInstance        *K8sOrchestrator
+	onceFork8sOrchestratorInstance sync.Once
 )
 
+// FSSConfigMapInfo contains details about the FSS configmap(s) present in all flavors
+type FSSConfigMapInfo struct {
+	featureStates      map[string]string
+	configMapName      string
+	configMapNamespace string
+}
+
 // K8sOrchestrator defines set of properties specific to K8s
 type K8sOrchestrator struct {
-	featureStates   map[string]string
+	supervisorFSS   FSSConfigMapInfo
+	internalFSS     FSSConfigMapInfo
 	informerManager *k8s.InformerManager
+	clusterFlavor   cnstypes.CnsClusterFlavor
+}
+
+// K8sGuestInitParams lists the set of parameters required to run the init for K8sOrchestrator in Guest cluster
```

Comment:
Can we use "TKG" or something instead of "Guest"

Reply:
We have been using Guest cluster till now in our code. We can use `Guest cluster` to be a generic term instead of being specific about TKG right?

Reply:
As we have been using GC everywhere and this is a minor issue, I am resolving this comment.